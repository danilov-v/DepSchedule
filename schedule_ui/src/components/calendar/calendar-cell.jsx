import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

import "./calendar.scss";

export function CalendarCell({ text, fluid }) {
  return (
    <div
      className={classnames(
        "calendar-cell d-flex align-items-center justify-content-center",
        {
          fluid,
        }
      )}
    >
      {text}
    </div>
  );
}

CalendarCell.propTypes = {
  text: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.node,
  ]),
  fluid: PropTypes.bool,
};

CalendarCell.defaultProps = {
  text: null,
  fluid: false,
};
