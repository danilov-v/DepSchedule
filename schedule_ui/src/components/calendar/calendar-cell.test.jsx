import React from "react";
import { shallow } from "enzyme";
import { CalendarCell } from "./calendar-cell";

describe("calendar-cell.jsx", () => {
  describe("<CalendarCell />", () => {
    test("should render and match snapshot", () => {
      const component = shallow(<CalendarCell text="25" />);

      expect(component).toMatchSnapshot();
    });

    test("should render and match snapshot with fluid", () => {
      const component = shallow(<CalendarCell text="25" fluid />);

      expect(component).toMatchSnapshot();
    });
  });
});
