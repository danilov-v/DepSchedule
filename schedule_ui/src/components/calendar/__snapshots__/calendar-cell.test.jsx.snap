// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calendar-cell.jsx <CalendarCell /> should render and match snapshot 1`] = `
<div
  className="calendar-cell d-flex align-items-center justify-content-center"
>
  25
</div>
`;

exports[`calendar-cell.jsx <CalendarCell /> should render and match snapshot with fluid 1`] = `
<div
  className="calendar-cell d-flex align-items-center justify-content-center fluid"
>
  25
</div>
`;
